### PR TITLE
 CI check for HDDS-4968

### PR DESCRIPTION
**Pull request by JacksonYao287**
## What changes were proposed in this pull request?

Back-port HDDS-4911 (List container by container state) to ContainerManagerV2

test whether this commit can pass the CI checks

## What is the link to the Apache JIRA

https://issues.apache.org/jira/browse/HDDS-4968

Please replace this section with the link to the Apache JIRA)

## How was this patch tested?

add new unit test

### CODE REVIEW

File: hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/container/ContainerManagerImpl.java [hunk 1]
```diff
@@ -152,15 +150,10 @@ public ContainerInfo getContainer(final ContainerID id)
...
-    // TODO: Remove the null check, startID should not be null. Fix the unit
-    //  test before removing the check.
-    final long start = startID == null ? 0 : startID.getId();
-    final List<ContainerID> containersIds =
-        new ArrayList<>(containerStateManager.getContainerIDs());
-    Collections.sort(containersIds);
-    return containersIds.stream()
-        .filter(id -> id.getId() > start).limit(count)
-        .map(ContainerID::getProtobuf)
+    final long start = startID.getId();
```
Comment (GlenGeng) [inline hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/container/ContainerManagerImpl.java:153]: The changes in this file seems not necessary to the back-port. We can create a refactor Jira to beautify the code later. 
We can reserve the `>=` change.

File: hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/server/SCMClientProtocolServer.java [hunk 2]
```diff
@@ -68,13 +68,7 @@
...
 import java.io.IOException;
 import java.net.InetSocketAddress;
-import java.util.Arrays;
-import java.util.List;
-import java.util.Map;
-import java.util.Collections;
-import java.util.ArrayList;
-import java.util.TreeSet;
-import java.util.Set;
+import java.util.*;
```
Comment (GlenGeng) [inline hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/server/SCMClientProtocolServer.java:71]: expand the *

File: hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/server/SCMClientProtocolServer.java [hunk 3]
```diff
@@ -362,7 +356,11 @@ private boolean hasRequiredReplicas(ContainerInfo contInfo) {
     }
     try {
       final ContainerID containerId = ContainerID.valueOf(startContainerID);
-      return scm.getContainerManager().getContainers(containerId, count);
+      if(null == state) {
+        return scm.getContainerManager().getContainers(containerId, count);
+      }
+      return scm.getContainerManager().getContainers(state).stream()
+          .limit(count).collect(Collectors.toList());
```
Comment (GlenGeng) [inline hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/server/SCMClientProtocolServer.java:363]: the `startContainerID` does not take effect here.

File: hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/container/ContainerManagerImpl.java (hunk 1 above)
Comment (JacksonYao287) [reply hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/container/ContainerManagerImpl.java]: ok ,done

File: hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/server/SCMClientProtocolServer.java (hunk 2 above)
Comment (JacksonYao287) [reply hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/server/SCMClientProtocolServer.java]: ok , midified

File: hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/server/SCMClientProtocolServer.java (hunk 3 above)
Comment (JacksonYao287) [reply hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/server/SCMClientProtocolServer.java:369]: thanks , has been fixed, please take a review

Comment (GlenGeng) [reply hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/server/SCMClientProtocolServer.java:370]: You seems miss sort on the stream.

Please check the indent for if and for as well.

Comment (JacksonYao287) [reply hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/server/SCMClientProtocolServer.java]: sorry, fixed

**Pull request merged**